indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+      {
+        InputStream stream;
+
+        @Override
+        public void open(InputStream in) throws IOException
+        {
+          this.stream = in;
+        }
+
+        @Override
+        public DataSegmentSplit deserialize(DataSegmentSplit dataSegmentSplit)
+            throws IOException
+        {
+          return new DataSegmentSplit(HadoopDruidConverterConfig.jsonMapper.readValue(stream, DataSegment.class));
+        }
+
+        @Override
+        public void close() throws IOException
+        {
+          stream.close();
+        }
+      };
+    }
+  }
+
+  public static class DataSegmentSplit extends InputSplit
+  {
+    private final DataSegment dataSegment;
+
+    public DataSegmentSplit(@NotNull DataSegment dataSegment)
+    {
+      this.dataSegment = Preconditions.checkNotNull(dataSegment, "dataSegment");
+    }
+
+    @Override
+    public long getLength() throws IOException, InterruptedException
+    {
+      return dataSegment.getSize();
+    }
+
+    @Override
+    public String[] getLocations() throws IOException, InterruptedException
+    {
+      return new String[]{};
+    }
+
+    protected DataSegment getDataSegment()
+    {
+      return dataSegment;
+    }
+  }
+
+  public static class ConfigInputFormat extends InputFormat<String, String>
+  {
+    @Override
+    public List<InputSplit> getSplits(final JobContext jobContext) throws IOException, InterruptedException
+    {
+      final HadoopDruidConverterConfig config = converterConfigFromConfiguration(jobContext.getConfiguration());
+      final List<DataSegment> segments = config.getSegments();
+      if (segments == null) {
+        throw new IOException("Bad config, missing segments");
+      }
+      return Lists.transform(
+          segments, new Function<DataSegment, InputSplit>()
+          {
+            @Nullable
+            @Override
+            public InputSplit apply(DataSegment input)
+            {
+              return new DataSegmentSplit(input);
+            }
+          }
+      );
+    }
+
+    @Override
+    public RecordReader<String, String> createRecordReader(
+        final InputSplit inputSplit, final TaskAttemptContext taskAttemptContext
+    ) throws IOException, InterruptedException
+    {
+      return new RecordReader<String, String>()
+      {
+        boolean readAnything = false;
+
+        @Override
+        public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext)
+            throws IOException, InterruptedException
+        {
+
+        }
+
+        @Override
+        public boolean nextKeyValue() throws IOException, InterruptedException
+        {
+          return !readAnything;
+        }
+
+        @Override
+        public String getCurrentKey() throws IOException, InterruptedException
+        {
+          return "key";
+        }
+
+        @Override
+        public String getCurrentValue() throws IOException, InterruptedException
+        {
+          readAnything = true;
```

Comment:
Required for `InputFormat<String, String>` which is required to get the proper `InputSplit` on the segments to have each segment be 1 Mapper